Modules/TOF/src/TOFMatchedTracks.cxx

```diff
@@ -0,0 +1,242 @@
+    ILOG(Info, Support) << "We found " << mTPCTOFMatches.size() << " TPC-only tracks matched to TOF" << ENDM;
+    if (mRecoCont.getTPCTOFTracks().size() != mTPCTOFMatches.size()) {
+      ILOG(Fatal, Support) << "Number of TPCTOF tracks (" << mRecoCont.getTPCTOFTracks().size() << ") differs from number of TPCTOF matches (" << mTPCTOFMatches.size() << ")" << ENDM;
+    }
+    // loop over TOF MatchInfo
+    for (const auto& matchTOF : mTPCTOFMatches) {
+      const auto& trk = mTPCTracks[matchTOF.getTrackIndex()];
+      mMatchedTracksPt[trkType::UNCONS]->Fill(trk.getPt());
+      mMatchedTracksEta[trkType::UNCONS]->Fill(trk.getEta());
+    }
+  }
+
+  // ITS-TPC-TOF
+  if (mRecoCont.isTrackSourceLoaded(GID::ITSTPCTOF)) { // this is enough to know that also ITSTPC was loades, see "initialize"
+    mITSTPCTracks = mRecoCont.getTPCITSTracks();
+    mITSTPCTOFMatches = mRecoCont.getITSTPCTOFMatches();
+    ILOG(Info, Support) << "We found " << mITSTPCTracks.size() << " ITS-TPC tracks" << ENDM;
+    ILOG(Info, Support) << "We found " << mITSTPCTOFMatches.size() << " ITS-TPC tracks matched to TOF" << ENDM;
+    // loop over TOF MatchInfo
+    for (const auto& matchTOF : mITSTPCTOFMatches) {
+      const auto& trk = mITSTPCTracks[matchTOF.getTrackIndex()];
+      mMatchedTracksPt[trkType::CONSTR]->Fill(trk.getPt());
+      mMatchedTracksEta[trkType::CONSTR]->Fill(trk.getEta());
+    }
+  }
+
+  auto creator = [this](auto& trk, GID, float, float) {
+    // Getting the tracks for the denominator of the efficiencies for TPC-TOF tracks;
+    // The RecoContainer will provide as TPCtracks only those not matched to TOF (lower
+    // quality), so we need to ask also for the TPCTOF to have the full set in the denominator
+    if constexpr (isTPCTrack<decltype(trk)>() || isTPCTOFTrack<decltype(trk)>()) {
+      // some quality cuts should be applied on the track.
+      // they might depend on the track type, so in case one needs to acces
+      // the correct track. For now, using an "if-statement" approach
+      // but it should be replaced with the official TrackCut class in O2.
+      // E.g. (till the TrackCut class is used):
+      int nMinTPCClusters = 0;
+      if constexpr (isTPCTrack<decltype(trk)>()) {
+        if (trk.getNClusters() < nMinTPCClusters) {
+          return true;
+        }
+      } else if constexpr (isTPCTOFTrack<decltype(trk)>()) {
+        const auto& tpcTrack = mTPCTracks[mTPCTOFMatches[trk.getRefMatch()].getTrackIndex()];
+        if (tpcTrack.getNClusters() < nMinTPCClusters) {
+          return true;
+        }
+      }
+      this->mInTracksPt[trkType::UNCONS]->Fill(trk.getPt());
+      this->mInTracksEta[trkType::UNCONS]->Fill(trk.getEta());
+    }
+    // In case of ITS-TPC-TOF, the ITS-TPC tracks contain also the ITS-TPC-TOF
+    if constexpr (isTPCITSTrack<decltype(trk)>()) {
+      this->mInTracksPt[trkType::CONSTR]->Fill(trk.getPt());
+      this->mInTracksEta[trkType::CONSTR]->Fill(trk.getEta());
+    }
```

Comment:
since you account the reference "pure" TPC tracks with some selections, for the consistency, the same cuts should be applied to the TPC tracks used in the matches.
This can access it via: 
````
auto tpcTr = this->mRecoCont.getTPCTrack( this->mRecoCont.getTPCContributorGID( GID)  );
````

Reply:
I am not sure I understand where you want to use this line. Maybe this can be taken care of by @njacazio / @noferini while I am away? Since it does not sound like a critical comment, if this PR is merged (I think we need @Barthelemy  or @knopers8 for this), they could do the change you propose in another one. If it is not merged, they can open a PR on my branch.

Reply:
I mean: if you want to normalize to TPC track passing certain cuts, this should be done for all TPC tracks, not only this which failed matching. So, one should get the TPC tracks contributing to TOF-matched tracks (with the line above) and accept the match only if it passes the same cuts. Sure, it can be done later, but I cannot merge in QC...

Reply:
Ah, yes, sure. But the part of the selections is not really implemented now, it was only to show how to access the track in the case of TPCTOF. 
I think that only @Barthelemy  and probably @knopers8 can merge...

Reply:
Hi @chiarazampolli I will handle it